src/commons/components/generate-form/generate-form.js

```diff
@@ -12,44 +12,42 @@
 
                 scope: {
                     formAttributes: "=",
-                    submitBtnName: "@"
+                    submitBtnName: "@",
+                    postUrl: "@",
+                    callBack: '&'
                 },
 
                 replace: false,
 
-                controller: function($scope, utils, config) {
-
-                    $scope.showClose = false;
+                controller: function($scope, utils) {
 
                     $scope.performAction = function() {
                         $scope.requestData = $scope.manipulateData();
 
-                        utils.takeAction($scope.requestData).then(function(jobObject) {
-                            alert("Volume is created successfully. and JOB-ID is - " + jobObject.job_id + " And Volume creation is " + jobObject.status);
+                        utils.takeAction($scope.requestData, $scope.postUrl).then(function(jobObject) {
+                            $scope.callBack(jobObject);
                         });
                         
                     };
 
                     $scope.manipulateData = function() {
-                        var keys = Object.keys($scope.formAttributes),
-                            len = keys.length,
-                            requestData = {},
-                            i;
+                        var requestData = {};
 
-                        for (i = 0; i < len; i++) {
-                            requestData[keys[i]] = $scope.formAttributes[keys[i]].value;
+                        for (var i = 0; i < $scope.formAttributes.length ; i++) {
```

Comment:
please refer http://stackoverflow.com/questions/8452317/do-loops-check-the-array-length-every-time-when-comparing-i-against-array-length